x-pack/plugins/security_solution/cypress/e2e/detection_rules/export_rule.cy.ts

```diff
@@ -68,27 +69,29 @@ describe('Export rules', () => {
   });
 
   it('exports only custom rules', function () {
-    const expectedNumberCustomRulesToBeExported = 1;
-    const totalNumberOfRules = expectedNumberCustomRulesToBeExported + totalNumberOfPrebuiltRules;
+    getPrepackagedRulesCount().then((prepackagedRulesCount) => {
+      const expectedNumberCustomRulesToBeExported = 1;
+      const totalNumberOfRules = expectedNumberCustomRulesToBeExported + prepackagedRulesCount;
 
-    loadPrebuiltDetectionRulesFromHeaderBtn();
+      loadPrebuiltDetectionRulesFromHeaderBtn();
 
-    selectAllRules();
-    bulkExportRules();
+      selectAllRules();
+      bulkExportRules();
 
-    cy.get(MODAL_CONFIRMATION_BODY).contains(
-      `${totalNumberOfPrebuiltRules} prebuilt Elastic rules (exporting prebuilt rules is not supported)`
-    );
+      cy.get(MODAL_CONFIRMATION_BODY).contains(
+        `${prepackagedRulesCount} prebuilt Elastic rules (exporting prebuilt rules is not supported)`
+      );
```

Comment:
Here we're getting the count before installing the rules. I guess, it works because `getPrepackagedRulesCount` returns the number of prebuilt rules (to be precise, the number of `security-rule` asset SOs) _available_ for installation rather than the number of installed prebuilt rules (the number of `alert` SOs).

I think this naming can be confusing and I'd rename `getPrepackagedRulesCount` to `getNumberOfAllAvailablePrebuiltRules` or smth like that.

Also, shouldn't we wait after calling `loadPrebuiltDetectionRulesFromHeaderBtn` until the installation is finished? I don't see any explicit check for this like `waitForPrebuiltDetectionRulesToBeLoaded`

Reply:
Yes, I see how that naming could be confusing. Changed `getPrepackagedRulesCount` to `getAvailablePrepackagedRulesCount`, hope that helps a bit.

> Also, shouldn't we wait after calling loadPrebuiltDetectionRulesFromHeaderBtn until the installation is finished?

Inside that method, we wait until the update button disappears, indicating the completion. So we should be good (until UI changes 🙂).